gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemIntegrationTest.java

```diff
@@ -1142,4 +1145,15 @@ public void testGetGcsPath() throws URISyntaxException {
     assertThrows(
         IllegalArgumentException.class, () -> myghfs.getGcsPath(new Path("gs://buck^et/object")));
   }
+
+  @Test
+  public void testInitializeCompatibleWithHadoopCredentialProvider() throws Exception {
+    Configuration config = loadConfig();
+
+    // This does not need to refer to a real bucket/path for the test.
+    config.set(HADOOP_SECURITY_CREDENTIAL_PROVIDER_PATH, "jceks://gs@foobar/test.jceks");
+
+    FileSystem.get(new URI(PUBLIC_BUCKET), config);
```

Comment:
Will this exercise the path where some provider is getting removed?

Reply:
Yes. I wrote this test before adding the call to `ProviderUtils`. I saw the test fail with the stack overflow error shown in #219. Then, after I added the call to `ProviderUtils`, the test passed.